reddit.py

```diff
@@ -0,0 +1,141 @@
+"""Reddit source code and datastore model classes.
+
+"""
+import logging
+
+from granary import reddit as gr_reddit
+from granary import source as gr_source
+from oauth_dropins import reddit as oauth_reddit
+from oauth_dropins.webutil.util import json_dumps, json_loads
+import webapp2
+from webob import exc
+
+import models
+import util
+
+
+class Reddit(models.Source):
+  """A Reddit account.
+
+  The key name is the username.
+  """
+  GR_CLASS = gr_reddit.Reddit
+  OAUTH_START_HANDLER = oauth_reddit.StartHandler
+  SHORT_NAME = 'reddit'
+  TYPE_LABELS = {
+    'post': 'submission',
+    'comment': 'comment',
+  }
+  TRANSIENT_ERROR_HTTP_CODES = ('404',)
```

Comment:
oh wow, reddit 404s are transient?! that's...odd and unfortunate.

Reply:
oh, this is probably not the case.  this was just a copy paste from twitter.py, and I didn't know what it did so I left it.  I've removed it now and it still seems to work.